packages/web3-utils/src/converters.ts

```diff
@@ -551,14 +561,23 @@ export const fromWei = (number: Numbers, unit: EtherUnits): string => {
  * ```
  */
 // todo in 1.x unit defaults to 'ether'
-export const toWei = (number: Numbers, unit: EtherUnits): string => {
+export const toWei = (number: Numbers, unit: EtherUnits | number): string => {
 	validator.validate(['number'], [number]);
 
-	const denomination = ethUnitMap[unit];
-
-	if (!denomination) {
-		throw new InvalidUnitError(unit);
+	let denomination;
+	if (typeof unit === 'string') {
+		denomination = ethUnitMap[unit];
+		if (!denomination) {
+			throw new InvalidUnitError(unit);
+		}
+	} else {
+		if (unit < 0 || !Number.isInteger(unit)) {
+			throw new InvalidIntegerError(unit);
+		}
+		
+		denomination = BigInt(10)**BigInt(unit);
```

Comment:
Sorry, this `**` will cause an issue. More at: https://github.com/web3/web3.js/pull/6506